src/partition/repart.c

```diff
@@ -5026,6 +5011,21 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return r;
 
+        if (arg_pretty > 0 ||
+            (arg_pretty < 0 && isatty(STDOUT_FILENO) > 0) ||
+            !FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF)) {
+
+                (void) context_dump_partitions(context, node);
+
+                if (arg_json_format_flags & JSON_FORMAT_OFF) {
+                        putc('\n', stdout);
+                        (void) context_dump_partition_bar(context, node);
+                        putc('\n', stdout);
+                }
+
+                fflush(stdout);
+        }
```

Comment:
should probably be a function of it's own actually, given that it's quite complex and main() mostly just dispatches complex stuff to other functions, and doesn#t do complex stuff on its own